oracle/src/feeds/gateio.rs

```diff
@@ -0,0 +1,73 @@
+use super::{get_http, PriceFeed};
+use crate::{config::CurrencyStore, currency::*, Error};
+use async_trait::async_trait;
+use clap::Parser;
+use reqwest::Url;
+use serde_json::Value;
+
+#[derive(Parser, Debug, Clone)]
+pub struct GateIoCli {
+    /// Fetch the exchange rate from gate.io
+    #[clap(long)]
+    gateio_url: Option<Url>,
+}
+
+pub struct GateIoApi {
+    url: Url,
+}
+
+impl Default for GateIoApi {
+    fn default() -> Self {
+        Self {
+            url: Url::parse("https://api.gateio.ws/api/v4").unwrap(),
+        }
+    }
+}
+
+fn extract_response<'a>(value: &'a Value) -> Option<&'a str> {
+    value.get(0)?.get("last")?.as_str()
+}
+
+impl GateIoApi {
+    pub fn from_opts(opts: GateIoCli) -> Option<Self> {
+        opts.gateio_url.map(Self::new)
+    }
+
+    pub fn new(url: Url) -> Self {
+        Self { url }
+    }
+
+    async fn get_exchange_rate(
+        &self,
+        currency_pair: CurrencyPair<Currency>,
+        currency_store: &CurrencyStore<Currency>,
+    ) -> Result<CurrencyPairAndPrice<Currency>, Error> {
+        // https://www.gate.io/docs/developers/apiv4/en/
+        let mut url = self.url.clone();
+        url.set_path(&format!("{}/spot/tickers", url.path()));
```

Comment:
I can't reach the docs page. Can you please document what kind of price we are fetching? I.e. daily opening, weighted average, etc. And for @philippb90 to check

Reply:
From the API it is described as the "last trading price", I'm not 100% sure what this means - perhaps from the last buy order?